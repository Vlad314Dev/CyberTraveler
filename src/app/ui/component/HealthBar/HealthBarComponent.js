import './HealthBarStyle';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

class HealthBarComponent extends PureComponent
{
    static propTypes = {
        p1Health: PropTypes.number.isRequired
    };

    render()
    {
        const { 
            p1Health,
        } = this.props;

        const healthBar = Array.from(Array(p1Health).keys());

        return (
            <div block="P1HealthBar" elem="Wrapper">
                {healthBar.map((value, index) => {
                    return <span 
                        block="P1HealthBar" 
                        elem="HealthElem"
                        key={ index }
                    ></span>
                })}
            </div>
        );
    }
}

export default HealthBarComponent;
