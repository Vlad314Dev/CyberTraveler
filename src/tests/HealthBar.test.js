/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import HealthBarComponent from '../app/ui/component/HealthBar/HealthBarComponent';

describe('HealthBar', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <HealthBarComponent 
                p1Health={ 5 }
            />
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <HealthBarComponent 
                p1Health={ 5 }
            />
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
