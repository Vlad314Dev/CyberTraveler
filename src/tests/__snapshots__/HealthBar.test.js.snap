// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthBar Should render layout 1`] = `
Array [
  <div
    className="P1HealthBar__Wrapper"
  >
    <span
      className="P1HealthBar__HealthElem"
    />
    <span
      className="P1HealthBar__HealthElem"
    />
    <span
      className="P1HealthBar__HealthElem"
    />
    <span
      className="P1HealthBar__HealthElem"
    />
    <span
      className="P1HealthBar__HealthElem"
    />
  </div>,
]
`;
